pkg/stream/integration/pg_pg_integration_transformer_test.go

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	pglib "github.com/xataio/pgstream/internal/postgres"
+	"github.com/xataio/pgstream/pkg/stream"
+)
+
+type transformerTestTableRow struct {
+	id             int
+	name           string
+	lastName       string
+	email          string
+	address        string
+	age            int
+	totalPurchases float64
+	customerID     string
+	birthDate      time.Time
+	isActive       bool
+	createdAt      int64
+	updatedAt      time.Time
+	gender         string
+}
+
+var createTableQuery = `CREATE TABLE %s(
+	id serial primary key,
+	name text,
+	last_name varchar(255),
+	email varchar(255),
+	address text,
+	age integer,
+	total_purchases double precision,
+	customer_id uuid,
+	birth_date date,
+	is_active bool,
+	created_at bigint,
+	updated_at timestamp with time zone,
+	gender varchar(255));`
+
+func Test_PostgresToPostgres_Transformer(t *testing.T) {
+	if os.Getenv("PGSTREAM_INTEGRATION_TESTS") == "" {
+		t.Skip("skipping integration test...")
+	}
+
+	cfg := &stream.Config{
+		Listener:  testPostgresListenerCfg(),
+		Processor: testPostgresProcessorCfgWithTransformer(pgurl),
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	runStream(t, ctx, cfg)
+
+	testTable := "pg2pg_integration_transformer_test"
+	execQuery(t, ctx, fmt.Sprintf(createTableQuery, testTable))
+
+	targetConn, err := pglib.NewConn(ctx, targetPGURL)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name string
+		rows []transformerTestTableRow
+	}{
+		{
+			name: "insert data",
+			rows: []transformerTestTableRow{
+				{
+					id:             1,
+					name:           "John",
+					lastName:       "Doe",
+					email:          "john.doe@example.com",
+					address:        "123 Main St",
+					age:            30,
+					totalPurchases: 1000.50,
+					customerID:     "123e4567-e89b-12d3-a456-426655440000",
+					birthDate:      time.Date(1990, 1, 1, 0, 0, 0, 0, time.UTC),
+					isActive:       true,
+					createdAt:      1672531200,
+					updatedAt:      time.Date(2023, 1, 1, 0, 0, 0, 0, time.UTC),
+					gender:         "male",
+				},
+			},
+		},
+	}
+
+	nextRowID := 1
+	for _, tc := range tests {
+		tc := tc
```

Comment:
You shouldn't need this anymore. It was required before go 1.22.
https://go.dev/blog/loopvar-preview